### PR TITLE
Add mutex around rng *rand.Rand to avoid data race on resume

### PR DESCRIPTION
**Pull request by jamesgroat**
Currently there is a data race when a consumer hits a backoff condition and then resumes. The resume via time.AfterFunc gets called from a different goroutine.

Note: this could also have been solved using one of the existing mutexes  (r.mtx or r.backoffMtx).

### CODE REVIEW

File: consumer.go [hunk 1]
```diff
@@ -1004,6 +1009,8 @@ func (r *Consumer) redistributeRDY() {
 		availableMaxInFlight = 1 - atomic.LoadInt64(&r.totalRdyCount)
 	}
 
+	r.rngMtx.Lock()
+	defer r.rngMtx.Unlock()
```
Comment (mreiferson) [inline consumer.go]: can we reduce the size of this critical path and put this lock around _just_ the call to `r.rng.Int()`?

**Comment by mreiferson** [comment]
thanks for this, just one minor comment!

**Comment by jamesgroat** [comment]
I've addressed the comment.

**Comment by mreiferson** [comment]
LGTM, do you mind squashing down to one commit?

File: consumer.go [hunk 2]
```diff
@@ -106,7 +106,8 @@ type Consumer struct {
 	channel string
 	config  Config
 
-	rng *rand.Rand
+	rngMtx sync.RWMutex
```
Comment (jehiah) [inline consumer.go]: `RWMutex` is designed for separate read and write locks. Since we only want global locking can you use `sync.Mutex` here?

**Comment by jamesgroat** [comment]
Squashed and changed to using sync.Mutex.

**Pull request merged**